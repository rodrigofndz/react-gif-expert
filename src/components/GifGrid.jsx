import { useEffect, useState } from "react";

import { GifItem } from "./GifItem";

import { useFetchGifs } from "../hooks/useFetchGifs";



export const GifGrid = ({category}) => {


    //CustomHook
    const {images, isLoading} = useFetchGifs(category);


    // const[images, setImages] = useState([]);

    // const getImages = async() => {
    //     const newImages = await getGifs(category);
    //     setImages(newImages);
    // }

    // useEffect(() => {
    //     getImages();
    // }, [])



    //OLD!!!!!!!
    // useEffect(() => {
    //     getGifs(category)
    //         .then(newImages => setImages(newImages));
    // }, [])
    

    
    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && <h2>Cargando...</h2>
            }
            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifItem 
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}
