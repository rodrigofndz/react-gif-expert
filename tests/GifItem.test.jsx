import { render } from "@testing-library/react";
import { GifItem } from "../src/components/GifItem";

describe('Pruebas <GifItem />', () => { 

    const image ={
        id: 1,
        title: 'Prueba',
        url: 'www.google.com'
    }

    test('should match snapshot', () => { 
        const {container} = render(<GifItem key={image.id} {...image}/>);
        expect(container).toMatchSnapshot();
     });
 });