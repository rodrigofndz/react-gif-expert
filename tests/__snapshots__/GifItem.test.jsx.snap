// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba"
      src="www.google.com"
    />
    <p>
      Prueba
    </p>
  </div>
</div>
`;
